pyspark_ai/pyspark_ai.py

```diff
@@ -113,7 +118,8 @@ def __init__(
             self._cache = None
         self._vector_store_dir = vector_store_dir
         self._vector_store_max_gb = vector_store_max_gb
-        self._encoding = encoding or tiktoken.get_encoding("cl100k_base")
+        if not create_deps_requirement_message:
+            self._encoding = encoding or tiktoken.get_encoding("cl100k_base")
```

Comment:
As per https://github.com/pyspark-ai/pyspark-ai/pull/162/files#r1361217186, we can now move the encoding into method `create_df`